### PR TITLE
Chef 10 & 11 deprecated, compatibility for >= 12.7

### PR DESCRIPTION
**Pull request by remeh**
- Chef < 12.7 compatibility has been removed
  - Now compatible with Chef 12.7, 13 and 14, cherry-picking some @martinisoft contributions in #101
  - Fix the CI accordingly.

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -1,24 +1,6 @@
-# encoding: utf-8
-
 # Latest release of mainline Chef versions here.
-%w(10 12).each do |tv|
+%w[12 12.7 13 14].each do |tv|
```
Comment (albertvaka) [inline Appraisals]: Should 12 be here in addition to 12.7?

Comment (albertvaka) [reply Appraisals]: I mean, because your PR says `Chef < 12.7 compatibility has been removed`.

Comment (remeh) [reply Appraisals]: Yes, the 12 entry actually targets `~> 12.0` which means latest 12: `12.22.5`

Comment (albertvaka) [reply Appraisals]: 👍

**Pull request merged**